components/scream/src/share/grid/abstract_grid.cpp

```diff
@@ -366,7 +366,7 @@ get_owners (const hview_1d<const gid_type>& gids) const
   //  - 1.a Figure where each local dof specs will be written in the linearly
   //        distributed global array
   for (int i=0; i<get_num_local_dofs(); ++i) {
-    const auto gid = m_dofs_gids_host[i];
+    const auto gid = m_dofs_gids_host[i]-get_global_min_dof_gid();
```

Comment:
While this works, I do not think it is the right thing to do. This is a GID, not an offset, and this fix might cause trouble down the road if something else in the lines below changes. Instead, I think we should subtract the min GID in the `pid_and_lid` lambda.

Reply:
Side comment, you should cache the value of the min GID. The routine involves global comm, and you don't want to do that multiple times!

Reply:
yes, I agree.  This is in a debug commit at the moment that I added so I could bring the changes back over to blake to do some unit testing.  But I already see this causes some hangs, which I don't fully understand TBH.  Still digging.